aws/aws_test.go

```diff
@@ -104,3 +105,44 @@ func TestAwsGetPrefixFromRootDeviceName(t *testing.T) {
 		assert.Equal(t, test.expectedPrefix, prefix)
 	}
 }
+
+type mockEC2Client struct {
+	ec2iface.EC2API
+	Vol *ec2.Volume
+}
+
+func (m mockEC2Client) CreateVolume(*ec2.CreateVolumeInput) (*ec2.Volume, error) {
+	return m.Vol, nil
+}
+
+func TestAwsCreate(t *testing.T) {
+	cases := []struct {
+		name           string
+		volumeTemplate interface{}
+		volResult      interface{}
+		expectedErr    error
+	}{
+		{
+			"illegal volume template from input is handled",
+			"I am anything but a valid ec2 volume template",
+			nil,
+			cloudops.NewStorageError(cloudops.ErrVolInval,
+				"Invalid volume template given", ""),
+		},
+	}
+	for _, c := range cases {
+		expectedRes, _ := c.volResult.(*ec2.Volume)
+		s := &awsOps{
+			ec2: &ec2Wrapper{
+				Client: mockEC2Client{Vol: expectedRes},
+			},
+		}
+		res, err := s.Create(c.volumeTemplate, nil, nil)
+		if c.expectedErr.Error() != err.Error() {
+			t.Errorf("%s failed, expected error: \n%v\n but got: \n%v", c.name, c.expectedErr, err)
+		}
```

Comment:
nit.. I think we should use assert.Equal
Something like assert.Equal(t, c.expectedErr, err, "<msg>")